src/Microsoft.AspNet.Mvc.TagHelpers/TagHelperOutputExtensions.cs

```diff
@@ -85,13 +84,22 @@ public static class TagHelperOutputExtensions
         {
             foreach (var attribute in tagBuilder.Attributes)
             {
-                if (!tagHelperOutput.Attributes.ContainsKey(attribute.Key))
+                if (!tagHelperOutput.Attributes.ContainsName(attribute.Key))
                 {
                     tagHelperOutput.Attributes.Add(attribute.Key, attribute.Value);
                 }
                 else if (attribute.Key.Equals("class", StringComparison.OrdinalIgnoreCase))
                 {
-                    tagHelperOutput.Attributes["class"] += " " + attribute.Value;
+                    TagHelperAttribute classAttribute;
+
+                    if (tagHelperOutput.Attributes.TryGetAttribute("class", out classAttribute))
+                    {
+                        classAttribute.Value += " " + attribute.Value;
```

Comment:
use `tagHelperOutput.Attributes["class"] = classAttribute.Value + " " + attribute.Value;` to remove other attributes named `class`